istioctl/cmd/istioctl/main.go

```diff
@@ -33,8 +37,28 @@ func main() {
 
 	rootCmd := cmd.GetRootCmd(os.Args[1:])
 
+	EnableKlogWithCobra()
+
 	if err := rootCmd.Execute(); err != nil {
 		exitCode := cmd.GetExitCode(err)
 		os.Exit(exitCode)
 	}
 }
+
+// EnableKlogWithCobra enables klog to work with cobra / pflags.
+// k8s libraries like client-go use klog.
+func EnableKlogWithCobra() {
+	klog.InitFlags(nil)
+	goflag.Parse()
+	// just add --v= flag to pflags.
+	gf := goflag.CommandLine.Lookup("v")
+	pflag.CommandLine.AddFlag(pflag.PFlagFromGoFlag(
+		&goflag.Flag{
+			Name:     "vklog",
+			Value:    gf.Value,
+			DefValue: gf.DefValue,
+			Usage:    gf.Usage + ". Like -v flag. ex: --vklog=9",
+		}))
+
+	klog.SetLogger(nil)
```

Comment:
If I understand the situation correctly:

Today we can enable "debug" logs from klog with  `--log_output_level=klog:debug`. This uses our logging system (json, stackdriver, etc). But it uses the default klog verbosity level which is not 9.

This pr allows setting the klog verbosity, but it doesn't use our logging system. I think if you remove this line it will allow --vklog to work, but it also requires --log_output_level=klog:debug. It seems preferable to use our logging system, but maybe we can improve the UX

I feel like this is something that is generally useful outside of istioctl - I have often wanted this in istiod. Maybe we should make it so it can be configured as part of istio/pkg? Not sure the UX we want here though (new flag, --log_output_level=klog:debug for most cases and env var to tune -v level, etc).

Reply:
yeah, once we have it in istioctl, we will add it to add binaries. 
In a previous version of the PR I had enabled it by mistake on istiod and is spews logs, but it is very useful.

Reply:
Can we at least remove `	klog.SetLogger(nil)` so it uses our logging system?